### PR TITLE
Allow invoker of `make check` to test a subset of packages.

### PR DESCRIPTION
**Pull request by hpidcock**
By passing `TEST_PACKAGE_LIST` env var set to the path of a file with sorted newline delimited package import paths, `make check` will only test that subset.

## QA steps

`make check` should still test all the packages it normally tests.

This should test only the `github.com/juju/juju/api/client/application` package.
`echo github.com/juju/juju/api/client/application > test.list && TEST_PACKAGE_LIST=test.list make check && rm test.list`

## Documentation changes

N/A

## Bug reference

N/A

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -279,7 +283,7 @@ test: run-tests
 run-tests:
 ## run-tests: Run the unit tests
 	$(eval TMP := $(shell mktemp -d $${TMPDIR:-/tmp}/jj-XXX))
-	$(eval TEST_PACKAGES := $(shell go list $(PROJECT)/... | grep -v $(PROJECT)$$ | grep -v $(PROJECT)/vendor/ | grep -v $(PROJECT)/acceptancetests/ | grep -v $(PROJECT)/generate/ | grep -v mocks))
+	$(eval TEST_PACKAGES := $(shell go list $(PROJECT)/... | sort | ([ -f $(TEST_PACKAGE_LIST) ] && comm -12 $(TEST_PACKAGE_LIST) - || cat) | grep -v $(PROJECT)$$ | grep -v $(PROJECT)/vendor/ | grep -v $(PROJECT)/acceptancetests/ | grep -v $(PROJECT)/generate/ | grep -v mocks))
```
Comment (tlm) [inline Makefile:286]: I think you can combine all the greps into a list with `[]` might read a bit better then?

**Review by tlm — APPROVED** [review]
lgtm

File: Makefile (hunk 1 above)
Comment (hpidcock) [reply Makefile:286]: I tried this and it was more messy but less verbose, so sticking with the current one.

**Comment by hpidcock** [comment]
$$merge$$

**Pull request merged**